lib/practice_exercises.rb

```diff
@@ -1,13 +1,36 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  index = -1
+
+  string.downcase!
+  string.gsub!(/[^A-Za-z]/, '')
+
+  string.each_char do |letter|
+    if letter == string[index]
+      index -=1
+    else
+      return false
+    end
+  end
+  return true
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
The loops here depend on the number of strings `n` and the length of the 1st string `m`, so this is O(nm) time complexity.  It also builds a prefix, so worst-case O(m) space complexity.